src/adfh/ADFH.c

```diff
@@ -2436,13 +2435,36 @@ void ADFH_Database_Open(const char   *name,
       set_error(FILE_OPEN_ERROR, err);
       return;
     }
+
+    /*
+      NOTE: Creation  order was set by  default  in CGNS 3.1.3, so a
+      CGNS file created by earlier versions will not have  this set.
+      Therefore, it should not be automatically assumed to be set in
+      H5Literate.
+    */
+
     gid = H5Gopen2(fid, "/", H5P_DEFAULT);
+
+    /* Obtain the group creation flags and check for link creation ordering. */
```

Comment:
`if (mode == ADFH_MODE_RDO) {`

Reply:
The problem is that it makes backward compatibility dependent on the mode the file is opened. CGNS should be able to read old files, regardless of how it is opened. It should be possible to open the file for modification and not write anything.

Reply:
We need to document that CGNS maintains one-way forward compatibility insofar as any file open and modified by, for example, 4.3.0 will be readable with 4.3.0+.  It can't be guaranteed the reverse 4.3.0- will be true.